### PR TITLE
 feat: add possibility to delete a draft expense

### PR DESCRIPTION
**Pull request by gca-axelor**
RM#72114

### CODE REVIEW

File: packages/apps/hr/src/features/expenseSlice.js [hunk 1]
```diff
@@ -229,6 +230,19 @@ export const fetchExpenseById = createAsyncThunk(
   },
 );
 
+export const deleteExpenseLine = createAsyncThunk(
```
Comment (vhu-axelor) [inline packages/apps/hr/src/features/expenseSlice.js:233]: Change name to `deleteExpense` and add it to `asyncFunctions-index.ts` file (export).

File: packages/apps/hr/src/i18n/en.json [hunk 2]
```diff
@@ -37,7 +37,7 @@
   "Hr_FromCity": "Starting city",
   "Hr_ToCity": "Arriving city",
   "Hr_Send": "Send",
-  "Hr_Validate": "Validate",
```
Comment (vhu-axelor) [inline packages/apps/hr/src/i18n/en.json:40]: Why did you delete this trad ?

File: packages/apps/hr/src/components/molecules/ExpenseDetailsValidationButton/ExpenseDetailsValidationButton.js [hunk 3]
```diff
@@ -57,8 +67,27 @@ const ExpenseDetailsValidationButton = ({expense, mode}) => {
...
+          title={I18n.t('Hr_Send')}
+          onPress={sendExpenseAPI}
+          width="45%"
+        />
+        <Button
+          title={I18n.t('Hr_Delete')}
+          onPress={deleteExpenseLineAPI}
+          width="45%"
+          color={Colors.errorColor}
+        />
```
Comment (vhu-axelor) [inline packages/apps/hr/src/components/molecules/ExpenseDetailsValidationButton/ExpenseDetailsValidationButton.js:88]: Maybe inverse order of buttons to match refuse and validate buttons (see below).

**Review by vhu-axelor — CHANGES_REQUESTED** [review]
Good job, works well! 🤩

File: packages/apps/hr/src/components/molecules/ExpenseDetailsValidationButton/ExpenseDetailsValidationButton.js (hunk 3 above)
Comment (lme-axelor) [reply packages/apps/hr/src/components/molecules/ExpenseDetailsValidationButton/ExpenseDetailsValidationButton.js:88]: And please add an icon on delete button

File: packages/apps/hr/src/screens/ExpenseListScreen.js [hunk 4]
```diff
@@ -159,7 +163,7 @@ const ExpenseListScreen = ({navigation}) => {
   );
 
   return (
-    <Screen removeSpaceOnTop={true}>
+    <FocusScreen removeSpaceOnTop={true} fetcher={refresh}>
```
Comment (lme-axelor) [inline packages/apps/hr/src/screens/ExpenseListScreen.js:166]: I think refresh should be on both types of expenses or at least on user's expenses as it's the most important on this screen.

**Review by vhu-axelor — APPROVED** [review]

**Review by vhu-axelor — APPROVED** [review]
Nice work!

File: packages/apps/hr/src/features/expenseSlice.js [hunk 5]
```diff
@@ -272,6 +309,13 @@ const expenseSlice = createSlice({
         manageTotal: true,
       },
     );
+    builder.addCase(silenceSearchExpense.pending, state => {
+      state.loadingMyExpense = false;
+    });
```
Comment (lme-axelor) [inline packages/apps/hr/src/features/expenseSlice.js:314]: remove this please, the objective is to have a silent refresh so we must not touch on loading attributes

File: packages/apps/hr/src/features/expenseSlice.js [hunk 6]
```diff
@@ -272,6 +309,13 @@ const expenseSlice = createSlice({
...
+    builder.addCase(silenceSearchExpense.fulfilled, (state, action) => {
+      state.myExpenseList = action.payload.myExpense;
+      state.expenseToValidateList = action.payload.expenseToValidate;
```
Comment (lme-axelor) [inline packages/apps/hr/src/features/expenseSlice.js:317]: You asked for 'returnTotalWithData' so action.payload.expenseToValidate contains data & total.
Also updating totalNumberExpenseToValidate could be great too

**Review by vhu-axelor — APPROVED** [review]

**Comment by lme-axelor** [comment]
I'm not sure the FocusScreen around a ScrollList is a good idea. Even if you created a silent search which was supposed to refresh data without the loading information, the FocusScreen trigger a refresh on the fetchData function so data is refresh two times : once with the silent search and once with the scrollList search which show the loading indicator. 
To avoid having to much refresh, maybe it's a better idea to call the search function of myExpense list in the then part of the delete reducer.
We can talk about it later if you had a idea of a better solution

File: packages/apps/hr/src/components/molecules/ExpenseDetailsValidationButton/ExpenseDetailsValidationButton.js [hunk 7]
```diff
@@ -57,13 +67,28 @@ const ExpenseDetailsValidationButton = ({expense, mode}) => {
     );
   }, [dispatch, mode, expense]);
 
+  const deleteExpenseAPI = useCallback(() => {
+    dispatch(deleteExpense({ExpenseId: expense.id, userId: user.id}));
+    navigation.navigate('ExpenseListScreen');
```
Comment (vhu-axelor) [inline packages/apps/hr/src/components/molecules/ExpenseDetailsValidationButton/ExpenseDetailsValidationButton.js:72]: I think it's better to use `navigation.pop()` here because this screen with this expense id will no longer exist so it's better to remove it from the stack

**Review by vhu-axelor — COMMENTED** [review]
Very nice work! 🤩

**Review by vhu-axelor — APPROVED** [review]
